src/main/java/core/basesyntax/serviceparser/FruitTransactionParserImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.serviceparser;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FruitTransactionParserImpl implements FruitTransactionParser {
+    @Override
+    public List<FruitTransaction> parse(List<String> lines) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        for (String line : lines) {
+            String[] parts = line.split(",");
+            if (parts.length != 3 || "type".equals(parts[0])) {
+                continue;
+            }
+            try {
+                FruitTransaction.Operation operation =
+                        FruitTransaction.Operation.valueOf(parts[0].toUpperCase());
```

Comment:
Ensure that the operation type in the input matches exactly with the enum values. If the input format changes or includes unexpected values, consider adding a mapping or validation mechanism to handle such cases more gracefully.